src/keeper.js

```diff
@@ -64,21 +64,29 @@ export default class keeper {
         const decimals27 = ethers.utils.parseEther('1000000000');
         let minProfitPercentage = ethers.utils.parseEther(Config.vars.minProfitPercentage);
 
+        const tab = auction.tab.div(decimal18);
         let calc = auction.price.mul(minProfitPercentage);
         let priceWithProfit = calc.div(decimal18);
 
 
         //adjusting lot to lotDaiValue
         let lotDaiValue = ethers.utils.parseEther(Config.vars.lotDaiValue).mul(decimal18);
         let minLot = lotDaiValue.div(auction.price.div(decimals9));
-        const lot = (auction.lot.toString());
+        let lot;
+
+        //checking for partial lot condition
+        let chost = clip._chost;
+        if(tab - lotDaiValue < chost) {
+          lot = auction.lot;
+        } else {
+          lot = minLot;
```

Comment:
This comment is more related to your previous change:  Shouldn't this value be the minimum of the auction lot, and the equivalent amount of `gem` configured in `lotDaiValue`, based on current auction price?  Seems like the user would configure `lotDaiValue` to limit maximum lot size.  The current code seems like it would try to take whatever that calculated amount of `gem` is, even if the auction itself is smaller.

Reply:
this code would evaluate if it needs to take the whole remaining lot or still use the minLot defined amount by `lotDaiValue`. And it takes the whole lot when `(tab - lotDaiValue < chost)` === true

Reply:
in the previous change I just implemented the `minLot` feature without taking into account the possibility of revert in case the auction tab is close to `chost` value and the tx can get reverted

Reply:
I realized it's fine calling `clip.take` with `amt` > `auction.lot`, eliminating my prior concern.